.github/workflows/ci.yml

```diff
@@ -2,189 +2,94 @@ name: Meshery Linkerd
+          # only-new-issues: true
+  tests:
+    name: tests
     runs-on: ubuntu-latest
-    needs: [lint, error_check, static_check, vet, sec_check, tests]
+    needs: [golangci]
     steps:
-    - name: Check out code
-      uses: actions/checkout@master
-      with:
-        fetch-depth: 1
-    - name: Setup Go
-      uses: actions/setup-go@v1
-      with:
-        go-version: ${{ secrets.GO_VERSION }}
-    - name: Cache node modules
-      uses: actions/cache@v1
-      with:
-        path: ~/go/pkg/mod
-        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-        restore-keys: |
-          ${{ runner.os }}-go-
-    - run: GOPROXY=direct GOSUMDB=off GO111MODULE=on go build .
+      - name: Check out code
+        uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ secrets.GO_VERSION }}
+      - name: Cache node modules
+        uses: actions/cache@v1
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Create cluster using KinD
+        uses: engineerd/setup-kind@v0.3.0
+        with:
+          version: "v0.7.0"
+      - run: |
+          export CURRENTCONTEXT="$(kubectl config current-context)"
+          echo "current-context:" ${CURRENTCONTEXT}
+          export KUBECONFIG="${HOME}/.kube/config"
+          echo "environment-kubeconfig:" ${KUBECONFIG}
+          GOPROXY=direct GOSUMDB=off GO111MODULE=on go test ./...
+
   docker:
     name: Docker build and push
     runs-on: ubuntu-latest
-    # needs: [build, build_release]
+    needs: [golangci,tests]
     steps:
     - name: Check out code
-      if: github.event_name != 'pull_request' && startsWith(github.ref, 'refs/tags/') && success()
       uses: actions/checkout@master
       with:
         fetch-depth: 1
     - name: Docker login
-      if: github.event_name != 'pull_request' && startsWith(github.ref, 'refs/tags/') && success()
-      uses: azure/container-actions/docker-login@master
+      uses: azure/docker-login@v1
       with:
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
-    - name: Docker build & tag
+    - name: Docker edge build & tag
+      if: startsWith(github.ref, 'refs/tags/') != true && success()
+      run: |
+        DOCKER_BUILDKIT=1 docker build --no-cache -t ${{ secrets.IMAGE_NAME }}:edge-latest --build-arg GIT_COMMITSHA=${GITHUB_SHA::7} .
+        docker tag ${{ secrets.IMAGE_NAME }}:edge-latest ${{ secrets.IMAGE_NAME }}:edge-${GITHUB_SHA::7}
+    - name: Docker edge push
+      if: startsWith(github.ref, 'refs/tags/') != true && success()
+      run: |
+        docker push ${{ secrets.IMAGE_NAME }}:edge-latest
+        docker push ${{ secrets.IMAGE_NAME }}:edge-${GITHUB_SHA::7}
```

Comment:
@Aisuko would you be so kind as to throw in a `:edge-${GITHUB_REF/refs\/tags\//}`, too?

Reply:
Here we can follow your idea, sir. The PR I follow Meshery format. As you can see with WIP. Is this would work for us?

Reply:
We should discuss clearly before merge the PR

Reply:
Done, and remove the other contexts. This PR only includes the CI changed.